pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1245,9 +1254,33 @@ protected void handleProducer(final CommandProducer cmdProducer) {
 
                     schemaVersionFuture.thenAccept(schemaVersion -> {
                         topic.checkIfTransactionBufferRecoverCompletely(isTxnEnabled).thenAccept(future -> {
-                            buildProducerAndAddTopic(topic, producerId, producerName, requestId, isEncrypted,
+                            CompletableFuture<Subscription> createInitSubFuture = new CompletableFuture<>();
+                            if (!Strings.isNullOrEmpty(initialSubscriptionName)
+                                    && !topic.getSubscriptions().containsKey(initialSubscriptionName)
+                                    && topic.isPersistent()) {
```

Comment:
Put `topic.isPersistent()` as the first or second param may be better

Reply:
You're right. Thanks!